checkstyle.xml

```diff
@@ -0,0 +1,351 @@
+<?xml version="1.0"?><!DOCTYPE module PUBLIC
```

Comment:
Based on [Google Java Code Style](https://google.github.io/styleguide/javaguide.html).

Reply:
Could you maybe update the android confluence docs to include this?

Reply:
Will definitely dp it, it will be the last subtask for this.

Reply:
Are you still planning to add this to the confluence doc? I noticed you edited the page and added the other links, but this would be useful to add too :)

Reply:
It is on the checkstlye page, but I can add the direct link for it.